plugin.video.ctuplinkrss/resources/lib/ctuplinkrss.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# c't uplink - The Audio Podcast Kodi Plug-in
+# by Achim Barczok, Heise Medien
+#
+# Diese Basisversion entspricht dem Beispiel in c't 19/2018
+# eine etwas schönere und umfangreiche Version 
+# finden Sie unter plugin.audio.ctuplink_audio
+
+import sys
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import feedparser
+
+
+def run():
+    #Selbst-Referenzierung fürs Plug-in
+    addon_handle = int(sys.argv[1])
+    
+    #Siedle Plug-in in den Bereich Audio
+    xbmcplugin.setContent(addon_handle, 'video')
+    
+    
+    #Parse ctuplink-Feed
+    d = feedparser.parse('https://blog.ct.de/ctuplink/ctuplinkvideohd.rss')
+    
+    #Liste mit allen Folgen aufbauen
+    listing = []
+    
+    for item in d['entries']:
+        title = item['title']
+        url = item.enclosures[0].href
```

Comment:
I'm not sure if this might result in HEAD requests for every item added here. I'm fairly sure that older versions of Kodi do this, but I'm not certain about the current status.

Reply:
I am not sure, what you mean by this.
Are you saying the *feedparser* library is making HEAD requests when I call `item.enclosures[0].href`?
Would it be better to get the url by accessing `item.links[0].href`?

Or is Kodi making the request when I list them using `xbmcplugin.addDirectoryItems`?

Could you maybe point me to the documentation that would help me debug the unnecessary HEAD requests?

Thank you!

Reply:
should be good to go...
i've tested the addon and didn't spot any excessive HEAD requests.